server/world/mcdb/provider.go

```diff
@@ -145,6 +147,83 @@ func (p *Provider) SaveSettings(s *world.Settings) {
 	p.saveDifficulty(s.Difficulty)
 }
 
+// LoadPlayerSpawnPosition loads the players spawn position stored in the level.dat from their UUID.
+func (p *Provider) LoadPlayerSpawnPosition(uuid uuid.UUID) (pos cube.Pos, exists bool, err error) {
+	data, err := p.db.Get(append([]byte("player_"), uuid[:]...), nil)
+	if err != nil {
+		return cube.Pos{}, false, err
+	}
+
+	var playerData map[string]string
+	if err := nbt.UnmarshalEncoding(data, &playerData, nbt.LittleEndian); err != nil {
+		return cube.Pos{}, false, err
+	}
+	if playerData["MsaID"] != uuid.String() || playerData["ServerId"] == "" {
+		return cube.Pos{}, false, fmt.Errorf("load player spawn: invalid data for player uuid: %v", uuid.String())
+	}
+	serverDB, err := p.db.Get([]byte(playerData["ServerId"]), nil)
+	if err != nil {
+		return cube.Pos{}, false, err
+	}
+
+	var serverData map[string]any
+	if err := nbt.UnmarshalEncoding(serverDB, &serverData, nbt.LittleEndian); err != nil {
+		return cube.Pos{}, false, err
+	}
+	x, y, z := serverData["SpawnX"], serverData["SpawnY"], serverData["SpawnZ"]
+	if x == nil || y == nil || z == nil {
+		return cube.Pos{}, false, nil
+	}
+	return cube.Pos{x.(int), y.(int), z.(int)}, true, nil
```

Comment:
These are of the type `int32`, so you might want to change them to `int(x.(int32))`.